doc syntax section into chapter, link from nixos manual (#241871)
NixOS/nixpkgs: nixos/doc/manual/contributing-to-this-manual.chapter.md

> The [DocBook] and CommonMark sources of the NixOS manual are in the [nixos/doc/manual](https://github.com/NixOS/nixpkgs/tree/master/nixos/doc/manual) subdirectory of the [Nixpkgs](https://github.com/NixOS/nixpkgs) repository.
> 
> See the [nixpkgs documentation syntax](https://nixos.org/manual/nixpkgs/unstable/#sec-contributing-markup).

Reviewer: ```suggestion
The sources of the NixOS manual are in the [nixos/doc/manual](https://github.com/NixOS/nixpkgs/tree/master/nixos/doc/manual) subdirectory of the [Nixpkgs](https://github.com/NixOS/nixpkgs) repository. It uses the same CommonMark variant as [Nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#sec-contributing-markup).
```

Author: We intentionally do not make any corrections other than the modifications that are necessary for linking. Corrections that have to do with docbook will be in another PR.

Reviewer: Well, I think just saying “see the nixpkgs documentation syntax” does not really connect – why should one see it?

Author: How about now, @jtojnar .
